ETL.md

```diff
@@ -0,0 +1,144 @@
+# ETL
+
+The ETL is a high level structure within Kubecost tasked with, as it name suggest, extracting cost and usage data from various sources (Prometheus, Thanos, Cloud Service Providers), transforming it into Asset and Allocation objects and loading it into caches that allow for fast querying.
+
+## Cloud Stores
+The ETL contains a Map of Cloud Stores, each of which represent an integration with a Cloud Service Provider. Each Cloud Store is responsible for the Cloud Asset and Reconciliation Pipelines which add Out-of-Cluster costs and Adjust Kubecost's estimated cost respectively via cost and usage data pulled from the Cloud Service Provider. Each Cloud Store has a unique identifier called the `ProviderKey` which varies depending on which Cloud Service Provider is being connected to and ensures that duplicate configurations are not introduced into the ETL. 
+
+- AWS: Account Id of the Master Billing Account
+- GCP: Project Id
+- Azure: Subscription Id
+
+The `ProviderKey` can be used as an argument for the endpoints related to the Cloud Processes (Cloud Assets and Reconciliation), to indicate that the specified operation should only be done on a single Cloud Store rather than all of them, which is the default behaviour. Additionally the Cloud Store keeps track of the Status of the Cloud Connection and Diagnostics for each of the Cloud Processes. The Cloud Connection Status is meant to be used as a tool in determining the health of the Cloud Connection that is the basis of each Cloud Store. The Cloud Connection Status has various failure states that are meant to provide actionable information on how to get your Cloud Connection running properly. These are the Cloud Connection Statuses:
```

Comment:
I think I get what Cloud Processes represents, but I would just leary of using too many technical terms that are less relevant to our end users.

Reply:
I was trying not to repeat "Cloud Assets and Reconciliation" a lot here, but if "Cloud Processes" is less clear I wont use it.